core/src/main/java/fj/Ord.java

```diff
@@ -514,4 +514,21 @@ public int compare(A o1, A o2) {
   public Comparator<A> toComparator() {
 	  return new OrdComparator();
   }
+
+  /**
+   * Given two Ord instances that compare the given type of object, chain them together so
+   * that the result of the first is returned if it is not EQ, otherwise the result
+   * of the second is returned.  This can be used to reuse an Ord between subclasses
+   * or implementations of an interface.
+   *
+   * @return An Ord instance combining the two provided Ord instances
+   */
+  public static <A> Ord<A> chain(Ord<? super A> ord1, Ord<? super A> ord2) {
```

Comment:
Without the variance annotation you can't directly reuse an Ord for a
superclass in a subclass, I think.  This is exactly why I created this
chain function, was to reuse Ord in subclasses.

On Mon, May 25, 2015, 1:43 AM JB Giraudeau <notifications@github.com> wrote:

> In core/src/main/java/fj/Ord.java
> <https://github.com/functionaljava/functionaljava/pull/139#discussion_r30969203>
> :
>
> > @@ -514,4 +514,21 @@ public int compare(A o1, A o2) {
> >    public Comparator<A> toComparator() {
> >  	  return new OrdComparator();
> >    }
> > +
> > +  /**
> > +   * Given two Ord instances that compare the given type of object, chain them together so
> > +   * that the result of the first is returned if it is not EQ, otherwise the result
> > +   * of the second is returned.  This can be used to reuse an Ord between subclasses
> > +   * or implementations of an interface.
> > +   *
> > +   * @return An Ord instance combining the two provided Ord instances
> > +   */
> > +  public static <A> Ord<A> chain(Ord<? super A> ord1, Ord<? super A> ord2) {
>
> In my understanding, functionaljava try to avoid the use of variance
> annotations (and subtyping as a whole), except for methods specifically
> created to deal with subtyping, like fj.Function.vary(F<? super A, ?
> extends B>).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/functionaljava/functionaljava/pull/139/files#r30969203>
> .
>